vm/interpreter/src/fvm.rs

```diff
@@ -164,62 +165,67 @@ impl<DB: Blockstore> Consensus for ForestExterns<DB> {
             );
         };
 
+        let mut fault_type: Option<ConsensusFaultType> = None;
+
         // (2) check for the consensus faults themselves
-        let mut cf: Option<ConsensusFault> = None;
 
         // (a) double-fork mining fault
         if bh_1.epoch() == bh_2.epoch() {
-            cf = Some(ConsensusFault {
-                target: *bh_1.miner_address(),
-                epoch: bh_2.epoch(),
-                fault_type: ConsensusFaultType::DoubleForkMining,
-            })
+            fault_type = Some(ConsensusFaultType::DoubleForkMining);
         };
 
         // (b) time-offset mining fault
         // strictly speaking no need to compare heights based on double fork mining check above,
         // but at same height this would be a different fault.
         if bh_1.parents() == bh_2.parents() && bh_1.epoch() != bh_2.epoch() {
-            cf = Some(ConsensusFault {
-                target: *bh_1.miner_address(),
-                epoch: bh_2.epoch(),
-                fault_type: ConsensusFaultType::TimeOffsetMining,
-            })
+            fault_type = Some(ConsensusFaultType::TimeOffsetMining);
         };
 
         // (c) parent-grinding fault
         // Here extra is the "witness", a third block that shows the connection between A and B as
         // A's sibling and B's parent.
         // Specifically, since A is of lower height, it must be that B was mined omitting A from its tipset
         if !extra.is_empty() {
+            // TODO: inspect decoding error case
             let bh_3 = BlockHeader::unmarshal_cbor(extra)?;
             if bh_1.parents() == bh_3.parents()
                 && bh_1.epoch() == bh_3.epoch()
                 && bh_2.parents().cids().contains(bh_3.cid())
                 && !bh_2.parents().cids().contains(bh_1.cid())
             {
-                cf = Some(ConsensusFault {
-                    target: *bh_1.miner_address(),
-                    epoch: bh_2.epoch(),
-                    fault_type: ConsensusFaultType::ParentGrinding,
-                })
+                fault_type = Some(ConsensusFaultType::ParentGrinding);
             }
         };
 
         // (3) return if no consensus fault
-        if cf.is_none() {
-            return Ok((cf, total_gas));
+        if fault_type.is_none() {
+            return Ok((ret, total_gas));
```

Comment:
`ret` is always `None` at this point, right?

Reply:
Yes.